### PR TITLE
feat(providers): add Groq + fix model-per-provider source of truth (#542)

### LINKED ISSUE #542: onboarding: verify quick start real mode works natively with all LLM providers
## Context
The quick start "real" flow lets users select a provider (OpenAI, Anthropic, Groq, Gemini) and run an actual optimization. We need to verify that each provider path works end-to-end out of the box — no extra setup, no hidden gotchas.

## Requirements
- Test the full quick start real flow with each provider:
  - OpenAI
  - Anthropic
  - Groq
  - Gemini
- Confirm: does each provider work natively, or are there extra dependencies, config steps, or edge cases?
- Document any provider-specific issues or limitations
- Fix or flag anything that breaks the seamless experience
- **The "what models to use per provider" logic should NOT be duplicated inside each provider individually.** It should be its own shared module/config that each provider calls into. Avoid per-provider model list duplication.

## Related
- #541 (full new-user journey)
- #524 (Mock button in provider selection)

### PR DESCRIPTION
**Pull request by EladTraignet**
## Summary
- Adds Groq as a fully supported provider (pattern detection, key validation, LangChain routing, model tiers)
- Fixes a pre-existing `"gemini"`→`"google"` key mismatch in `_FALLBACK_MODELS`/`_MODEL_TIERS` that was silently falling back to OpenAI models for Google
- Eliminates model-list duplication in the walkthrough: `07_multi_provider` (real + mock) now derives model names from `get_models_for_tier()` in core

## Changes
- `traigent/providers/validation.py` — Groq pattern (`llama-|llama3-|gemma2?-|mixtral-`), known models, `_validate_groq()`, LiteLLM prefix map entry
- `traigent/integrations/providers.py` — Groq tiers/fallbacks; rename `"gemini"` keys to `"google"`; update Google balanced/quality to include `gemini-2.0-flash`
- `pyproject.toml` — add `langchain-groq>=0.2.0` dependency
- `walkthrough/real/07_multi_provider.py` — use `get_models_for_tier()` per provider; add `ChatGroq` routing; add `GROQ_API_KEY` to mock fallback check
- `walkthrough/mock/07_multi_provider.py` — same model-list refactor; adds Groq models to mock config space

## Test plan
- [ ] Run `walkthrough/real/07_multi_provider.py` with `OPENAI_API_KEY` only → should run OpenAI models, skip others
- [ ] Run with `ANTHROPIC_API_KEY` only → should run Anthropic models
- [ ] Run with `GOOGLE_API_KEY` only → should run Gemini models
- [ ] Run with `GROQ_API_KEY` only → should run Llama/Groq models
- [ ] Run with all four keys → should run all 8 models
- [ ] Run with no keys → should fall back to mock

Closes #542

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: examples/core/text-to-sql/run.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+        async def main() -> None:
+            trials = 12 if not MOCK else 12  # covers all 8 config combos + oversampling
+            result = await generate_sql.optimize(algorithm="grid", max_trials=trials)
+
+            print("\n" + "=" * 60)
+            print("OPTIMIZATION COMPLETE")
+            print("=" * 60)
+            print(f"Best config : {result.best_config}")
+            print(f"Best score  : {result.best_score:.3f}")
```
[the remark this answers is not in the document]
Comment (EladTraignet) [reply examples/core/text-to-sql/run.py:226]: Fixed — removed the no-op ternary; the line now simply reads `trials = 12`.
